src/blockchain/Blockchain.ts

```diff
@@ -141,28 +200,135 @@ export default class Blockchain implements IBlockchainApi {
     return this.api.rpc.chain.subscribeNewHeads(listener)
   }
 
+  /**
+   * [ASYNC] Signs the SubmittableExtrinsic with the given identity.
+   *
+   * @param identity The [[Identity]] to sign the Tx with.
+   * @param tx The unsigned SubmittableExtrinsic.
+   * @returns Signed SubmittableExtrinsic.
+   *
+   */
   public async signTx(
     identity: Identity,
     tx: SubmittableExtrinsic
   ): Promise<SubmittableExtrinsic> {
     const nonce = await this.getNonce(identity.address)
-    const signed: SubmittableExtrinsic = identity.signSubmittableExtrinsic(
+    const signed: SubmittableExtrinsic = await identity.signSubmittableExtrinsic(
       tx,
-      nonce.toHex()
+      nonce
     )
     return signed
   }
 
+  /**
+   * [ASYNC] Submits a signed [[SubmittableExtrinsic]] with exported function [[submitSignedTx]].
+   * Handles recoverable errors by re-signing and re-sending the tx up to two times.
+   *
+   * Transaction fees will apply whenever a transaction fee makes it into a block, even if extrinsics fail to execute correctly!
+   *
+   * @param identity The [[Identity]] to potentially re-sign the Tx with.
+   * @param tx The [[SubmittableExtrinsic]] to be submitted. Most transactions need to be signed, this must be done beforehand.
+   * @param opts Criteria for resolving/rejecting the promise.
+   * @returns A promise which can be used to track transaction status.
+   * If resolved, this promise returns the eventually resolved [[SubmittableResult]].
+   */
+  public async submitSignedTx(
+    identity: Identity,
+    tx: SubmittableExtrinsic,
+    opts?: SubscriptionPromiseOptions
+  ): Promise<SubmittableResult> {
+    const options = parseSubscriptionOptions(opts)
+    return submitSignedTx(tx, options)
+      .catch(async (reason: Error) => {
+        if (reason.message === 'Recoverable') {
+          return submitSignedTx(await this.reSignTx(identity, tx), options)
+        }
+        throw reason
+      })
```

Comment:
Wow that's a really easy retry! Love it! But maybe refactor the async closure into it's own variable and have

```ts
const retry = async (...) => {...}
return submitSignedTx(tx, options).catch(retry).catch(retry)
```

Reply:
Good suggestion, might even make it possible in the future to have the user specify how often they want to retry.